src/index.js

```diff
@@ -1,2 +1,106 @@
+/*
+ * This script assumes AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are exported as environment variables
+ */
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const http = require('http');
+const isProduction = process.env.NODE_ENV === 'production';
+const outputFile = 'blog-posts-data.json';
+const s3Config = {
+  bucket: 'data.pvdgeeks.org',
+  key: 'blogpost'
+};
 
-console.log('Hello! The current date and time is: ' + Date()); // eslint-disable-line no-console
\ No newline at end of file
+// expose handler for Lambda
+exports.run = run;
+
+if (!isProduction) {
+  run();
+}
+
+function run() {
+  const resolveAllPromises = isProduction ? resolveBlogPostsDataS3 : resolveBlogPostsDataLocal;
```

Comment:
if we're only make one request, then the name should probably change away from a plural, right?  So maybe something like `promiseResolver`